core/vm/contracts_zkevm.go

```diff
@@ -737,14 +886,22 @@ func (c *bls12381Pairing_zkevm) Run(input []byte) ([]byte, error) {
 }
 
 // bls12381MapG1 implements EIP-2537 MapG1 precompile.
-type bls12381MapG1_zkevm struct{}
+type bls12381MapG1_zkevm struct {
+	enabled bool
+}
 
 // RequiredGas returns the gas required to execute the pre-compiled contract.
 func (c *bls12381MapG1_zkevm) RequiredGas(input []byte) uint64 {
+	if !c.enabled {
+		return 0
+	}
 	return params.Bls12381MapG1Gas
 }
 
 func (c *bls12381MapG1_zkevm) Run(input []byte) ([]byte, error) {
+	if !c.enabled {
```

Comment:
instead of that, maybe we should just not remove these precompiles to the previous fork id (pre forkID7) and keep them always enabled? will that break anything?

Reply:
I think because zkevm does not support these precompiles now,  it will be run with ErrExecutionReverted. So, maybe we should keep the execute result same,  and disabed some precompiles. 
According to this:
https://github.com/0xPolygonHermez/cdk-erigon/issues/37#issuecomment-1891244596